src/plugins/audio-waveform/audio-waveform-editor.js

```diff
@@ -0,0 +1,136 @@
+
+  const { sourceType, sourceUrl, width } = content;
+
+  const changeContent = newContentValues => {
+    const newContent = { ...content, ...newContentValues };
+    const isInvalidSourceUrl = newContent.sourceType !== IMAGE_SOURCE_TYPE.internal && validation.validateUrl(newContent.sourceUrl, t).validateStatus === 'error';
+    onContentChanged(newContent, isInvalidSourceUrl);
+  };
+
+  const handleSourceTypeChange = event => {
+    const { value } = event.target;
+    changeContent({ sourceType: value, sourceUrl: '' });
+  };
+
+  const handleSourceUrlChange = event => {
+    const { value } = event.target;
+    changeContent({ sourceUrl: value });
+  };
+
+  const handleInternalUrlChange = url => {
+    changeContent({ sourceUrl: urlToStorageLocationPath(url) });
+  };
+
+  const handleCreateWaveform = () => {
+    setIsWaveformGeneratorDialogVisible(true);
+  };
+
+  const handleWaveformGeneratorDialogSelect = newInternalSourceUrl => {
+    setIsWaveformGeneratorDialogVisible(false);
+    changeContent({
+      sourceType: IMAGE_SOURCE_TYPE.internal,
+      sourceUrl: urlToStorageLocationPath(newInternalSourceUrl)
+    });
+  };
+
+  const handleWaveformGeneratorDialogCancel = () => {
+    setIsWaveformGeneratorDialogVisible(false);
+  };
+
+  const handleWidthChange = value => {
+    changeContent({ width: value });
+  };
+
+  return (
+    <div>
+      <Form layout="horizontal">
+        <Divider plain>{t('chooseImageDividerText')}</Divider>
+        <FormItem label={t('common:source')} {...formItemLayout}>
+          <RadioGroup value={sourceType} onChange={handleSourceTypeChange}>
+            <RadioButton value={IMAGE_SOURCE_TYPE.external}>{t('common:externalLink')}</RadioButton>
+            <RadioButton value={IMAGE_SOURCE_TYPE.internal}>{t('common:internalCdn')}</RadioButton>
+          </RadioGroup>
+        </FormItem>
+        {sourceType === 'external' && (
+          <FormItem label={t('common:externalUrl')} {...formItemLayout} {...validation.validateUrl(sourceUrl, t)} hasFeedback>
+            <Input value={sourceUrl} onChange={handleSourceUrlChange} />
+          </FormItem>
+        )}
+        {sourceType === 'internal' && (
+          <FormItem label={t('common:internalUrl')} {...formItemLayout}>
+            <div className="u-input-and-button">
+              <Input
+                addonBefore={CDN_URL_PREFIX}
+                value={sourceUrl}
+                onChange={handleSourceUrlChange}
+                />
+              <ResourcePicker
+                url={storageLocationPathToUrl(sourceUrl)}
+                onUrlChange={handleInternalUrlChange}
+                />
+            </div>
+          </FormItem>
+        )}
+        <Divider plain>{t('generateImageDividerText')}</Divider>
+        <FormItem {...formItemLayoutWithoutLabel}>
+          <Button type="primary" onClick={handleCreateWaveform}>
+            {t('Wellenform aus Audio-Datei erstellen')}
```

Comment:
Wrong text here 😅

Reply:
Oh my! I forgot to add the translation. Fixed! Thanks for spotting it.